backend/src/utils/rcon.js

```diff
@@ -0,0 +1,69 @@
+const { Rcon } = require('rcon-client');
+const { Command } = require('../sql/models');
+
+module.exports = async (scheduled) => {
+  const cronId = scheduled.getTime();
+
+  try {
+    // If any commands need to be run, set the cron timeStamp
+    // on the command
+    const updates = await Command.update({ cronId }, {
+      where: {
+        status: 'READY',
+        cronId: null,
+      },
+    });
+
+    // If no commands were updated, nothing will be running
+    if (updates.length === 0) return;
+
+    // Get all these updated commands
+    const commands = await Command.findAll({
+      where: {
+        status: 'READY',
+        cronId,
+      },
+    });
+
```

Comment:
I think we could save a lot of db calls if we immediately set all the found commands to running instead of doing it each time (in line 43). Then when completing the command (setting to finished in line 49) we can retain this state locally as well. When reaching the error in line 58 we could set all commands that are still RUNNING status back to READY in the db in one query. Not sure if this will be necessary, but depending how many commands we get could save a lot of time due to reduced db calls.

Reply:
Addressed in [1927c0e](https://github.com/aidan-brown/Charity-Stream-Web-App/pull/77/commits/1927c0e8fad4ff45373d31176bc0a904f3a2a88c), good call out